### PR TITLE
[BugFix] Fix NPE when the field name includes a dot

### PR DESCRIPTION
**Pull request by predator4ann**
When the field name contains a dot, NPE will be encountered because consider it a child field of structType.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [x] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by miomiocat — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 1]
```diff
@@ -132,10 +132,17 @@ private static Type getResultType(String columnName, IcebergContext context) {
...
-            String[] paths = qualifiedName.split("\\.");
             Type type = context.getSchema();
-            for (String path : paths) {
-                type = type.asStructType().fieldType(path);
+            Types.StructType structType = type.asStructType();
+            type = structType.fieldType(qualifiedName);
+            if (null != type) {
+                return type;
+            }
+            if (qualifiedName.contains(".")) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:141]: why does your column name has dot?

Comment (predator4ann) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:141]: For example, we have a business wide table with columns named `event_time.string` and `event_time.int`, which are different event tracking logs with the same field name but different data types

Comment (predator4ann) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:141]: The reason why I changed this part of the code is to consider that if iceberg originally supported it, then starrocks could also be able to support it

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:140]: please add an sql test case

Comment (predator4ann) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:140]: done

**Review by miomiocat — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**